homeassistant/components/heos/__init__.py

```diff
@@ -1,19 +1,20 @@
 """Denon HEOS Media Player."""
-
 import asyncio
 import logging
 
 import voluptuous as vol
 
 from homeassistant.components.media_player.const import (
     DOMAIN as MEDIA_PLAYER_DOMAIN)
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_HOST, EVENT_HOMEASSISTANT_STOP
 import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.discovery import async_load_platform
 from homeassistant.helpers.typing import ConfigType, HomeAssistantType
 
-DOMAIN = 'heos'
-REQUIREMENTS = ['aioheos==0.4.0']
+from .config_flow import format_title
+from .const import DATA_CONTROLLER, DOMAIN
+
+REQUIREMENTS = ['pyheos==0.2.0']
```

Comment:
This change should be done in a separate PR from config entry support

Reply:
It's a minimal change, not sure I agree.

Reply:
It is hard to see when so many other things are happening in the PR as well

Reply:
There isn't that much going on here and it's already been reviewed with tests written.  The config entry part represents < 40 lines of the overall code and the overhead of breaking that out when it would rewrite code that was already being rewritten with the lib change isn't justified given the overhead of PRs.

Reply:
I think @Kane610 has a point. This PR could have been broken up into smaller pieces. Eg, requirement change, config entries, device_info.

At this point I've already looked at it though. I'll need to go over the tests again before merge.

Reply:
Agree it could have, but my point is we need to balance small batches with the overhead of a PR, especially if those subsequent PRs will rewrite code that is already changing.  That creates wasted review work and increases our build queue.  Smaller is not always better--it's a balance.

Reply:
Well @andrewsayre you should have known better ;). Martin is already knee deep into this so I'll just end it here :)

Reply:
You can argue that it is not black and white but a multi-hundred line PR is probably not the place to do it. Working on the code for days makes you know it well so you think that it is not as complex as it really is.

Sure it takes work but there is big value in reworking code changes into smaller steps. You will find and fix issues (not just bugs) to make the result easier to review. This saves review time and improves the code.

Smaller PRs can be reviewed by different people so that will also speed up things, even if there is some overlap. Furthermore, keeping it small helps with fatigue so reviewers will catch more things.

Regarding `async_unload_entry()`, I do agree that it would be strange to leave that one out :)